File: demos/csharp/dsa/trees/HeapDemo.cs
using System;
using System.Collections.Generic;

public class HeapDemo
{
    public static void Main()
    {
        Console.WriteLine("=== C# PriorityQueue<TElement, TPriority> ===");
        var pq = new PriorityQueue<string, int>();
        pq.Enqueue("Low Priority", 30);
        pq.Enqueue("High Priority", 10);
        pq.Enqueue("Medium Priority", 20);

        string first = pq.Dequeue();
        if (first != "High Priority") throw new Exception("PriorityQueue failed");

        Console.WriteLine($"Dequeued highest priority element: {first}");
        Console.WriteLine("C# PriorityQueue tests passed successfully.");
    }
}

File: demos/java/dsa/trees/HeapDemo.java
public class HeapDemo {
    public static void main(String[] args) {
        System.out.println("=== Java PriorityQueue (Min/Max Heap) ===");
        java.util.PriorityQueue<Integer> minHeap = new java.util.PriorityQueue<>();
        minHeap.offer(50);
        minHeap.offer(15);
        minHeap.offer(30);

        if (minHeap.peek() != 15) throw new RuntimeException("Heap peek failed");
        if (minHeap.poll() != 15) throw new RuntimeException("Heap poll failed");
        if (minHeap.poll() != 30) throw new RuntimeException("Heap poll failed");
        if (minHeap.poll() != 50) throw new RuntimeException("Heap poll failed");

        System.out.println("Java PriorityQueue tests passed successfully.");
    }
}
